### PR TITLE
Fixed #5317: added a toolbar icon to open JSON in Keyboard Settings

### LINKED ISSUE #5317: [settings] Add a toolbar button to open `keymaps.json` in Keyboard Settings UI
On the Keyboard Settings UI page, it would be nice to add a toolbar button that directs to the `keymaps.json` file when clicked.

### PR DESCRIPTION
**Pull request by fangnx**
Fixed #5317.

In `Keyboard Settings` UI, added a toolbar button that opens `keymaps.json` file.

- [x] Register [CQ](https://dev.eclipse.org/ipzilla/show_bug.cgi?id=20146)

### CODE REVIEW

File: packages/keymaps/src/browser/keybindings-widget.tsx [hunk 1]
```diff
@@ -38,6 +38,8 @@ export interface CellData {
     highlighted: boolean,
 }
 
+export const KEYBINDING_WIDGET_ID = 'keybindings.view.widget';
```
Comment (vince-fugnitto) [inline packages/keymaps/src/browser/keybindings-widget.tsx]: Why was this necessary?

File: packages/keymaps/src/browser/keymaps-frontend-contribution.ts [hunk 2]
```diff
@@ -81,4 +86,17 @@ export class KeymapsFrontendContribution extends AbstractViewContribution<Keybin
...
+    registerToolbarItems(toolbar: TabBarToolbarRegistry): void {
+        toolbar.registerItem({
+            id: KeymapsCommands.OPEN_KEYMAPS_JSON.id,
+            command: KeymapsCommands.OPEN_KEYMAPS_JSON.id,
+            tooltip: 'Open Keyboard Shortcuts in JSON'
+        });
+    }
+
+    protected isKeybindingWidget(widget: Widget | undefined = this.tryGetWidget()): boolean {
+        return widget instanceof KeybindingWidget && widget.id === KEYBINDING_WIDGET_ID
```
Comment (vince-fugnitto) [inline packages/keymaps/src/browser/keymaps-frontend-contribution.ts]: I believe you can simply use `KeybindingWidget.ID` instead of `KEYBINDING_WIDGET_ID`.

File: packages/keymaps/src/browser/keymaps-frontend-contribution.ts [hunk 3]
```diff
@@ -36,19 +39,20 @@ export namespace KeymapsCommands {
     export const OPEN_KEYMAPS_JSON: Command = {
         id: 'keymaps:openJson',
         category: 'Settings',
-        label: 'Open Keyboard Shortcuts (JSON)'
+        label: 'Open Keyboard Shortcuts (JSON)',
+        iconClass: 'theia-open-file-icon'
```
Comment (vince-fugnitto) [inline packages/keymaps/src/browser/keymaps-frontend-contribution.ts]: I'm not sure this is the appropriate icon for the toolbar.
I'd expect this one even if it requires a CQ, it will be useful for other areas in Theia.

<div align='center'>

Screen Shot 2019-06-06 at 11 33 36 AM


</div>

File: packages/keymaps/src/browser/keymaps-frontend-contribution.ts (hunk 2 above)
Comment (fangnx) [reply packages/keymaps/src/browser/keymaps-frontend-contribution.ts]: Oh that was my mistake... Will correct it

File: packages/keymaps/src/browser/keymaps-frontend-contribution.ts (hunk 3 above)
Comment (fangnx) [reply packages/keymaps/src/browser/keymaps-frontend-contribution.ts]: Yep, a distinguishable icon would be nice

File: packages/keymaps/src/browser/keymaps-frontend-contribution.ts [hunk 4]
```diff
@@ -63,6 +67,7 @@ export class KeymapsFrontendContribution extends AbstractViewContribution<Keybin
         });
         commands.registerCommand(KeymapsCommands.OPEN_KEYMAPS_JSON, {
             isEnabled: () => true,
+            isVisible: widget => this.isKeybindingWidget(widget),
```
Comment (vince-fugnitto) [inline packages/keymaps/src/browser/keymaps-frontend-contribution.ts]: There is a bug where we cannot run the command `Settings: Open Keyboard Shortcuts (JSON)` from the command palette unless the keybindings-widget is currently opened.

**Reproduce:**
1. attempt to open the JSON file using <kbd>F1</kbd> → `Settings: Open Keyboard Shortcuts (JSON)`
2. notice that the command is **not present** in the quick-commands menu
3. open the keybindings-widget
4. repeat step 1
5. notice that the command is now **present** in the quick-commands menu

Comment (fangnx) [reply packages/keymaps/src/browser/keymaps-frontend-contribution.ts]: Thanks Vince, I will look into it rn

File: packages/core/src/browser/icons/edit-json-dark.svg [hunk 5]
```diff
@@ -0,0 +1,6 @@
+<!--Copyright (c) Microsoft Corporation. All rights reserved.-->
+<!--Copyright (C) 2018 Ericsson and others.-->
```
Comment (vince-fugnitto) [inline packages/core/src/browser/icons/edit-json-dark.svg]: Don't forget to register a [CQ](https://github.com/theia-ide/theia/wiki/Registering-CQs) for both copied files :)

File: packages/keymaps/src/browser/keymaps-frontend-contribution.ts (hunk 4 above)
Comment (fangnx) [reply packages/keymaps/src/browser/keymaps-frontend-contribution.ts]: Just fixed this by separating command palettte & toolbar commands.

**Comment by vince-fugnitto** [comment]
The CQ has been approved so merging! :)
